ngraph/core/reference/src/runtime/reference/fft.cpp

```diff
@@ -0,0 +1,635 @@
+                    const auto output_outer_strides = get_lengths(output_strides, outer_axes);
+                    const auto reversed_input_shape = reverse_shape(input_data_shape);
+                    const auto input_fft_lengths = get_lengths(reversed_input_shape, fft_axes);
+                    const auto input_strides = compute_strides(reversed_input_shape);
+                    const auto input_fft_strides = get_lengths(input_strides, fft_axes);
+                    const auto input_outer_strides = get_lengths(input_strides, outer_axes);
+
+                    result.fft_axes = fft_axes;
+                    result.fft_lengths = fft_lengths;
+                    result.fft_strides = fft_strides;
+                    result.outer_strides = outer_strides;
+                    result.output_fft_strides = output_fft_strides;
+                    result.output_outer_strides = output_outer_strides;
+                    result.input_fft_lengths = input_fft_lengths;
+                    result.input_fft_strides = input_fft_strides;
+                    result.input_outer_strides = input_outer_strides;
+                    result.fft_rank = fft_rank;
+                    result.fft_size = fft_size;
+                    result.outer_size = outer_size;
+                    result.buffer_size = buffer_size;
+
+                    return result;
+                }
+            }
+
+            // Calculation of FFT
+            void fft(const float* input_data,
+                     const Shape& input_data_shape,
+                     const int64_t* axes_data,
+                     const Shape& axes_data_shape,
+                     float* fft_result,
+                     const Shape& output_shape,
+                     FFTKind fft_kind)
+            {
+                const complex_type* complex_input_data_ptr =
+                    reinterpret_cast<const complex_type*>(input_data);
+                complex_type* complex_output_ptr = reinterpret_cast<complex_type*>(fft_result);
+
+                const auto info = get_info_for_calculation(
+                    input_data_shape, axes_data, axes_data_shape, output_shape);
+                const auto& fft_axes = info.fft_axes;
+                const int64_t fft_rank = info.fft_rank;
+                const auto& fft_lengths = info.fft_lengths;
+                const auto& fft_strides = info.fft_strides;
+                const int64_t fft_size = info.fft_size;
+
+                if (fft_size <= 0)
+                {
+                    return;
+                }
+
+                std::vector<complex_type> data(fft_size);
+                std::vector<complex_type> buffer(info.buffer_size);
+
+                const auto& output_fft_strides = info.output_fft_strides;
+                const auto& outer_strides = info.outer_strides;
+                const int64_t outer_size = info.outer_size;
+
+                const auto& output_outer_strides = info.output_outer_strides;
```

Comment:
The same comment for here. Why do you need to have so much references to constant variables?

Reply:
We need it for brevity, to write `fft_axes` instead of `info.fft_axes`, `fft_lengths` instead of `info.fft_lengths`, etc